src/udev/udev-builtin-net_id.c

```diff
@@ -656,6 +657,23 @@ static int names_mac(struct udev_device *dev, struct netnames *names) {
         unsigned int i;
         unsigned int a1, a2, a3, a4, a5, a6;
 
+        /* Some kinds of devices tend to have hardware addresses
+         * that are impossible to use in an iface name.
+         */
+        s = udev_device_get_sysattr_value(dev, "type");
+        if (!s)
+                return EXIT_FAILURE;
+        i = strtoul(s, NULL, 0);
+        switch (i) {
+        /* The persistent part of a hardware address of an InfiniBand NIC
+         * is 8 bytes long. We cannot fit this much in an iface name.
+         */
+        case ARPHRD_INFINIBAND:
+                return -EINVAL;
```

Comment:
Maybe just write this as
```C
if (i == ARPHRD_INFINIBAND)
    return -EINVAL;
```
?

Reply:
I chose the switch statement mainly because we might eventually end up adding new cases for that in the future, if more link layer technologies with long HW addresses appear.

Reply:
That's true. If you want to keep that, it's OK, especially that the surrounding code uses the same pattern. Nevertheless, nowadays if we had more choices we'd just write:
```C
if (IN_SET(i, ARPHRD_INFINIBAND, ARPHRD_FOO, ...)
     ...
```